### DISCUSSION TITLE
Question regarding adding observables to flavio

### DISCUSSION THREAD

**Discussion by blackstonep** [flav-io/flavio#285] [Q&A] [answered]
I am wondering how one goes about adding an observable along with its theoretical expression in flavio. One example that I want to include is BR(tau->omegamu), for example, where the V = rho and phi branching ratios are already included. Is the correct strategy in this specific case to amend the br_tauvl.py file to account for the relevant theoretical expression, or is this a more treacherous task? I have noticed the Observable.from_function method in the case of forming an observable composed of already-implemented ones, have not found information on writing a fresh observable in this manner. 

For the record, I am also hoping to implement tau -> lP decays as well as tau -> lP1P2 (two pseudoscalar) processes. The latter doesn't have any related instances already included in flavio -- is this because of a fundamental difficulty in handling these decays? 

Thank you for your time. (Also, apologies if this is not the optimal method of posing questions.)

**Comment by peterstangl** [comment, marked as answer]
> I am wondering how one goes about adding an observable along with its theoretical expression in flavio. One example that I want to include is BR(tau->omegamu), for example, where the V = rho and phi branching ratios are already included. Is the correct strategy in this specific case to amend the br_tauvl.py file to account for the relevant theoretical expression, or is this a more treacherous task? I have noticed the Observable.from_function method in the case of forming an observable composed of already-implemented ones, have not found information on writing a fresh observable in this manner.

For the general implementation of new observables, you can find some information at https://flav-io.github.io/docs/add-observable.html

If there are already similar observables implemented and the new observable would require just small modifications, it might make indeed sense to add the new observable to the existing files.

> For the record, I am also hoping to implement tau -> lP decays as well as tau -> lP1P2 (two pseudoscalar) processes. The latter doesn't have any related instances already included in flavio -- is this because of a fundamental difficulty in handling these decays?

If the theory prediction for these observables is available in the literature, including all necessary hadronic input etc., there should be no fundamental difficulty implementing them in flavio.

> Thank you for your time. (Also, apologies if this is not the optimal method of posing questions.)

The issue system on GitHub is actually the perfect place for asking questions and discussing the implementation of new observables :slightly_smiling_face:
